### ISSUE TITLE
After importing a tensorflow/keras converted to ONNX model, the input is (0, 64, 64, 1) with an incorrect batch size.

### ISSUE THREAD

**Issue by jordi-ander** [Unity-Technologies/barracuda-release#92]
After importing the model, we see the input as (0, 64, 64, 1) but should be (1, 64, 64, 1). The model was tested after conversion on another machine, not running in Unity, with a (1, 64, 64, 1) input and executed as expected.
<img width="652" alt="image" src="https://user-images.githubusercontent.com/67168222/95143460-f4ae0c00-072a-11eb-90d3-ac2d67a6002d.png">
We also see the output as (-1, -1, -1, -1), however after reading through some of the already posted issues, I understand that the (-1, -1, -1, -1) output is not necessarily an issue.

When we pass a (1, 64, 64, 1) tensor to the model worker (containing a Texture2D as input), we get the following error 
<img width="1192" alt="image" src="https://user-images.githubusercontent.com/67168222/95143753-cc72dd00-072b-11eb-86ba-6d2757c0d56d.png">

Does anyone know what steps we should take to ensure the input is correct when the model is imported into Unity?

Thanks!

**Comment by amirebrahimi** [comment]
Can you share the code you are using to supply the input to the worker?

**Comment by AlexRibard** [comment]
Hi @jordi-ander !
`(0, 64, 64, 1)` is mostly bad naming convention on our end. 0 means *, ie variable number of batches. It won't cause any issue in executing your model.
I do see that you are using a Softmax with a specified axis. That is unfortunately unsupported in current versions and will be in the upcoming one :)

As for the runtime error, that seems strange. Could you compare with the docs that you are executing correctly?
https://github.com/Unity-Technologies/barracuda-release/blob/release/1.0.0/Documentation~/ModelExecution.md

As @amirebrahimi mentioned, if you could provide code/model it would help us narrow down the issue

**Comment by mantasp** [comment]
@jordi-ander, any chance that your project has Resources folder with some TextAssets in it?

**Comment by jordi-ander** [comment]
This is how I'm creating my worker, creating the tensor, and executing everything.

I have the following as global variables:
`public NNModel modelAssest;`
`private Model m_RuntimeModel;`
<img width="672" alt="image" src="https://user-images.githubusercontent.com/67168222/95242393-f546b100-07c3-11eb-9c8a-3fbc09fb0d4d.png">

@AlexRibard Thanks for your explanation! Do you know if there is an anticipated release date for the new version?

**Comment by AlexRibard** [comment]
That seems correct... how is `m_RuntimeModel` created?
Also a few issues I see in your code.
* you are using  a CSharp worker with a Texture, unfortunately atm this will not work.  So either use a GPU worker or download the texture via `ReadPixels`.
* also beware of `Resize` call on texture
https://docs.unity3d.com/ScriptReference/Texture2D.Resize.html
It does not actually do interpolation. For that, I would suggest the following:
-- use `Upsample2D` on the tensor created from the texture
-- use a `Material` and blit your texture onto it's render-target.
(cf this example project)
https://github.com/AlexRibard/Barracuda-U-2-NetTest/blob/master/Assets/Scripts/Inference.cs

**Comment by jordi-ander** [comment]
@AlexRibard 
Do any if any of the CPU worker types work with a RenderTexure?
Also, do you know the release date for the new version of Barracuda?

**Comment by jordi-ander** [comment]
@AlexRibard Do you know the release date for the new version of Barracuda? 

Secondly, we've gotten an error with ReduceMean when using version 9 & 12. Is there a version we should use to avoid this error?
<img width="1182" alt="Screen Shot 2020-10-28 at 2 36 02 PM" src="https://user-images.githubusercontent.com/67168222/97499714-97713900-192b-11eb-8c0a-f4e6b3a7497b.png">

**Comment by AlexRibard** [comment]
sorry, notification got lost.
No CPU workers do not work with `RenderTexture` for that you'll need to call `GetPixels` from a Texture
(cf https://answers.unity.com/questions/27968/getpixels-of-rendertexture.html) and create the tensor from that

**Comment by AlexRibard** [comment]
as for the new release 1.2.0 we intend on publishing it this week.
For the `keedims` in `ReduceMean` you can use 1.1.2 it's implemented there.
Which model are you using atm? I can test it out on 1.2.0

**Comment by jordi-ander** [comment]
No problem! Thanks for your notes!

I updated to the newest version (1.2.0) and we no longer have either of the ReduceMean or the Softmax errors. However, the input shapes have changed. I have two models in my asset folder and both their inputs were (1, 64, 64, 1) in version 1.0.3. In this newer version they are (1, 1, 0, 1, 1, 64, 64, 1) and (1, 1, 0, 1, 1, 64, 1, 64).

After updating, I reimported both models and still had the incorrect inputs. 

<img width="629" alt="Screen Shot 2020-10-30 at 4 50 20 PM" src="https://user-images.githubusercontent.com/67168222/97765852-1653a680-1ad1-11eb-9a02-f4139a1a89f8.png">

**Comment by jordi-ander** [comment]
@AlexRibard Have you heard about the above error after updating to version 1.2.0?

**Comment by AlexRibard** [comment]
I have now! I don't think I have your models. Can you send me the model for me to check?
You can email me at barracuda-support@unity3d.com if you don't want to post your model here

**Comment by jordi-ander** [comment]
@Alex thanks! I'm emailing you a version of our model now!

**Comment by AlexRibard** [comment]
fix has been pushed in latest
